REVERSE_PROXY.md

```diff
@@ -0,0 +1,418 @@
+
+### 2. Create Password File
+
+```bash
+# For nginx
+sudo htpasswd -c /etc/nginx/fdo-mgmt.passwd admin
+
+# For Apache  
+sudo htpasswd -c /etc/httpd/fdo-mgmt.passwd admin
+# Enter password when prompted
+```
+
+**Important notes:**
+- Record the password you enter, as it will be hashed in the password file and cannot be recovered. You'll need this password for API authentication.
+- The `-c` flag creates a new password file, overwriting any existing file. For adding additional users later, omit the `-c` flag: `sudo htpasswd /etc/nginx/fdo-mgmt.passwd another_user`
+
+### 3. Firewall Configuration
+
+```bash
+# Allow HTTPS traffic (required for both)
+sudo firewall-cmd --add-service=https --permanent
+sudo firewall-cmd --reload
+```
+
+## Nginx Configuration
+
+### Configuration File
+
+The nginx configuration below creates two separate virtual hosts (one for Manufacturing APIs, one for Owner APIs) with the following security features:
+
+- **TLS Encryption**: Forces HTTPS with modern TLS protocols and secure ciphers
+- **Basic Authentication**: Requires username/password for `/api/v1/` endpoints  
+- **Security Headers**: Adds headers to protect against common attacks
+- **Health Check Access**: Allows unauthenticated access to `/health` for monitoring
+- **Request Blocking**: Returns 404 for any other requests
+
+Create `/etc/nginx/conf.d/fdo-mgmt.conf`:
+
+```nginx
+# FDO Manufacturing Management API
+server {
+    listen 443 ssl http2;
+    server_name fdo-mfg.example.com;
+    
+    # TLS Configuration
+    ssl_certificate /path/to/your/cert.pem;
+    ssl_certificate_key /path/to/your/key.pem;
+    ssl_protocols TLSv1.2 TLSv1.3;
+    ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384;
+    ssl_prefer_server_ciphers off;
+    
+    # Security Headers
+    add_header X-Frame-Options DENY;
+    add_header X-Content-Type-Options nosniff;
+    add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
+    
+    # Protect management APIs
+    location /api/v1/ {
+        auth_basic "FDO Management";
+        auth_basic_user_file /etc/nginx/fdo-mgmt.passwd;
+        
+        proxy_pass http://127.0.0.1:8038;
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+        
+        # Timeouts for API calls
+        proxy_connect_timeout 30s;
+        proxy_send_timeout 30s;
+        proxy_read_timeout 30s;
+    }
+    
+    # Health check (no auth required)
+    location /health {
+        proxy_pass http://127.0.0.1:8038;
+        proxy_set_header Host $host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+    }
+    
+    # Block everything else
+    location / {
```

Comment:
Should we provide unauthenticated access to the /fdo/101 API subtree as well?  IIUC FDO doesn't require auth on these endpoints (doing its own auth in the protocol).  Might make sense since the server ports (8038, etc) are firewalled off.

Opinions?